sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -113,16 +117,32 @@ class FileStreamSource(
   // Visible for testing and debugging in production.
   val seenFiles = new SeenFilesMap(maxFileAgeMs, fileNameOnly)
 
-  private var allFilesForTriggerAvailableNow: Seq[(SparkPath, Long)] = _
+  private var allFilesForTriggerAvailableNow: Seq[NewFileEntry] = _
 
   metadataLog.restore().foreach { entry =>
     seenFiles.add(entry.sparkPath, entry.timestamp)
   }
   seenFiles.purge()
 
-  logInfo(s"maxFilesPerBatch = $maxFilesPerBatch, maxFileAgeMs = $maxFileAgeMs")
+  logInfo(s"maxFilesPerBatch = $maxFilesPerBatch, " +
+    s"maxBytesPerBatch = $maxBytesPerBatch, maxFileAgeMs = $maxFileAgeMs")
+
+  private var unreadFiles: Seq[NewFileEntry] = _
 
-  private var unreadFiles: Seq[(SparkPath, Long)] = _
+  /**
+   *  Split files into a selected/unselected pair according to a total size threshold.
+   */
+  private def takeFilesUntilMax(files: Seq[NewFileEntry], maxSize: Long):
+    (Seq[NewFileEntry], Seq[NewFileEntry]) = {
+    var idx = 0
+    var totalSize = 0L
+    val (bFiles, usFiles) = files.span { case NewFileEntry(_, size, _) =>
+      idx += 1
+      totalSize += size
+      idx == 1 || totalSize <= maxSize
```

Comment:
If we change the type of `maxSize` from `Int` to `Long`, line 141 and 142 should be revised in order to avoid overflow.

Reply:
As I understand overflow might happen if a sum of input file sizes(in bytes) will be bigger that Long.MaxValue in bytes which is ~ 9PB. Should we consider that a possibility?

Reply:
Yes, of course, we had better have a safe code from the beginning for the future. We can check that before doing `totalSize += size`, right?

Reply:
I used Math.addExact to deal with overflow here in another place. Let's me know if that approach is acceptable. I also found Spark's MathUtils. Not 100% certain which way is the better.

Reply:
This is better than the silent overflow. However, this looks insufficient to me because this new code enforces Apache Spark community to add a new migration documentation about this kind of new regression .

Although I understand why you think the overflow is unlikely and reluctant to change the logic, the Apache Spark community shares this code among the world-wide users. We cannot merge this kind of regression. 

To simply put, when we able to handle handle this gracefully without any exception and regression, why do we choose this logic and expose the users the risk of production job failures due to `ArithmeticException`, @MaxNevermind ?

Reply:
@dongjoon-hyun 
I misunderstood and was under impression that throwing exception is expected behavior.
I adjusted the the code, no exceptions now, file split stops before `Long` overflow, please check it out.